packages/components/src/internal/components/picklist/actions.ts

```diff
@@ -0,0 +1,233 @@
+                    keyType: 'AutoIncrementInteger',
+                    description : picklist.Description,
+                    category: picklist.Category,
+                };
+                if (picklist.name !== domain.get('name')) {
+                    options = Object.assign(options, {name: picklist.name});
+                }
+                saveDomain(
+                    domain,
+                    PICKLIST,
+                    options,
+                    picklist.name
+                ).then(savedDomain => {
+                    resolve(picklist);
+                }).catch(errorDomain => {
+                    console.error(errorDomain.domainException);
+                    reject(errorDomain.domainException);
+                });
+            })
+            .catch(reason => {
+                console.error(reason);
+                reject(reason);
+            });
+    })
+}
+
+export function addSamplesToPicklist(listName: string, selectionKey?: string, sampleIds?: string[] ) : Promise<InsertRowsResponse> {
+    let rows = List<any>();
+    if (selectionKey) {
+        getSelected(selectionKey).then(response => {
+            response.selected.forEach(id => {
+                rows = rows.push({ SampleID: id});
+            });
+            return insertRows({
+                schemaQuery: SchemaQuery.create("lists", listName),
+                rows
+            });
+        });
+    }
+    else {
+        sampleIds.forEach(id => {
+            rows = rows.push({ SampleID: id})
+        });
+        return insertRows({
+            schemaQuery: SchemaQuery.create("lists", listName),
+            rows
+        });
+    }
+}
+
+export interface PicklistDeletionData {
+    numDeletable: number,
+    numNotDeletable: number,
+    numShared: number,
+    deletableLists: PicklistModel[],
+}
+
+export function getPicklistDeleteData(model: QueryModel, actions: Actions, user: User) : Promise<PicklistDeletionData> {
+    return new Promise((resolve, reject) => {
+
+        const columnString = 'Name,listId,category,createdBy';
+        getSelectedData(model.schemaName, model.queryName, [...model.selections], columnString, undefined, undefined,'ListId')
+            .then(response => {
+                    const { data } = response;
+                    let numDeletable = 0;
+                    let numNotDeletable = 0;
+                    let numShared = 0;
+                    let deletableLists = [];
+                    data.valueSeq().forEach(row => {
+                        const picklist = new PicklistModel(flattenValuesFromRow(row.toJS(), row.keySeq().toArray()));
+
+                        if (picklist.isDeletable(user)) {
+                            if (picklist.isPublic()) {
+                                numShared++;
+                            }
+                            deletableLists.push(picklist);
+                        } else {
+                            numNotDeletable++;
+                        }
```

Comment:
I don't think it is likely to happen, but looking at the implementation for `isDeletable`, it looks like it is technically possible for a private list to not be deletable. If so, should there be two if statement blocks here: 1 for getting the count of private vs public and a 2nd for getting the count of deletable and non-deletable?